recipes/fmt/all/conanfile.py

```diff
@@ -64,65 +58,63 @@ def config_options(self):
-                  destination=self._source_subfolder, strip_root=True)
-
-    def _configure_cmake(self):
-        if self._cmake:
-            return self._cmake
-        self._cmake = CMake(self)
-        self._cmake.definitions["FMT_DOC"] = False
-        self._cmake.definitions["FMT_TEST"] = False
-        self._cmake.definitions["FMT_INSTALL"] = True
-        self._cmake.definitions["FMT_LIB_DIR"] = "lib"
-        if self._has_with_os_api_option:
-            self._cmake.definitions["FMT_OS"] = self.options.with_os_api
-        self._cmake.configure(build_folder=self._build_subfolder)
-        return self._cmake
+        get(self, **self.conan_data["sources"][str(self.version)], destination=self.source_folder, strip_root=True)
 
     def build(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
+        apply_conandata_patches(self)
         if not self.options.header_only:
-            cmake = self._configure_cmake()
+            cmake = CMake(self)
+            # FIXME : https://github.com/conan-io/conan/issues/11476
+            cache_entries = {
+                "FMT_DOC": "False",
+                "FMT_TEST": "False",
+                "FMT_INSTALL": "True",
+                "FMT_LIB_DIR": "lib"
+            }
+            if self._has_with_os_api_option:
+                cache_entries["FMT_OS"] = self.options.with_os_api
+            cmake.configure(variables=cache_entries)
             cmake.build()
 
+    @staticmethod
+    def _rm_folder(folder):
+        shutil.rmtree(folder, ignore_errors=True)
+
     def package(self):
-        self.copy("LICENSE.rst", dst="licenses", src=self._source_subfolder)
+        copy(self, pattern="*LICENSE.rst", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
         if self.options.header_only:
-            self.copy("*.h", dst="include", src=os.path.join(self._source_subfolder, "include"))
+            copy(self, pattern="*.h", src=os.path.join(self.source_folder, "include"), dst=os.path.join(self.package_folder, "include"))
         else:
-            cmake = self._configure_cmake()
+            cmake = CMake(self)
             cmake.install()
-            tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
-            tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
-            tools.rmdir(os.path.join(self.package_folder, "share"))
+            self._rm_folder(os.path.join(self.package_folder, "lib", "cmake"))
+            self._rm_folder(os.path.join(self.package_folder, "lib", "pkgconfig"))
+            self._rm_folder(os.path.join(self.package_folder, "res"))
+            self._rm_folder(os.path.join(self.package_folder, "share"))
 
     def package_info(self):
         self.cpp_info.names["cmake_find_package"] = "fmt"
         self.cpp_info.names["cmake_find_package_multi"] = "fmt"
         self.cpp_info.names["pkg_config"] = "fmt"
```

Comment:
I also disagree with writing these useless changes that could be a simple comment. In addition to that, they lead us to think "there are missing set_properties"

Reply:
I agree with that, this is a source of confusion if you are not very familiar with the repo... I would totally translate all of those to a comment.

Reply:
do you mean this line is useless:
```
self.cpp_info.names["pkg_config"] = "fmt"
```
or what exactly?

Reply:
Yes, because the name by default is the name of the package, so the three cpp_info.names lines are redundant. If i recall correctly this was the way of marking that the upstream used that target names or something like that?

Reply:
I remember @SpaceIm had some concerns removing it.
basically, it might be useless, but it appears to be harmless.
and if CCI contributors feel more convenient with it, I am fine leaving it.

Reply:
But introducing `a=1` 100 times in the code is also harmless, you know what I mean...

Reply:
we might agree on that, but it's out of scope of that PR. the scope is to ensure V2 compatibility. removing these lines doesn't improve it, it's a side change and goes to an another PR.

Reply:
I don't want to argue again on this topic. It's very important to track upstream CMake names (I can't remember the number of broken recipes because it was missing).